### PR TITLE
Anomalies now have observer-visible countdowns

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Anomalies now have observer-visible countdowns to their detonation.
/:cl:

- Behind the scenes, moved the detonation code from the event to the
anomalies themselves. May be some differences of lifespan, given we're
going from event ticks to world.time, but meh.

### CODE REVIEW

**Comment by XDTM** [comment]
Since you're working on anomalies, could you make the anomaly cores say which anomaly they came from, please? They used to be all the same, but now they have separate research levels.

**Comment by Cyberboss** [comment]
@optimumtact Restart travis, the error is a lie

File: code/game/objects/effects/anomalies.dm [hunk 1]
```diff
@@ -9,10 +9,20 @@
 	luminosity = 3
 	var/movechance = 70
 	var/obj/item/device/assembly/signaler/anomaly/aSignal = null
+	var/area/impact_area
+
+	var/lifespan = 990
+	var/death_time
+
+	var/countdown_colour
```
Comment (coiax) [inline code/game/objects/effects/anomalies.dm]: I was hoping that @ChangelingRain would improve my terrible color choices at some point in the future.

**Comment by XDTM** [comment]
thanks

**Pull request merged**